src/JustEat.StatsD/Transport/Data.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+
+namespace JustEat.StatsD
+{
+    public static class DataExtensions
+    {
+        public static ReadOnlySpan<byte> GetSpan(this in Data src)
+        {
+            if (src.Array == null)
+            {
+                return src.Span;
+            }
+            return src.Array.AsSpan();
+        }
+
+        public static byte[] GetArray(this in Data src)
+        {
+            if (src.Array == null)
+            {
+                return src.Span.ToArray();
```

Comment:
What about `Array.Empty<byte>()`?

Reply:
Not sure I understood your point, so if Data block was created out of span, and consumer asked for an array, return an empty array?

This code will function correctly but will allocate on the heap. If we want to prevent allocation we could throw an exception.